### PR TITLE
DROOLS-4485: [DMN Designer] Data Types - Add visual indication for showing the current drag level

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.redhat.com/browse/DROOLS-4485

### CODE REVIEW

**Comment by karreiro** [comment]
@lclay2 I've updated this PR with a small enhancement :-) See:

Before:


After:

**Comment by karreiro** [comment]
@danielzhe @jomarko Could you please take a look?

**Comment by lclay2** [comment]
@karreiro Wow, good catch, thanks!

**Comment by karreiro** [comment]
Hey @lclay2 now the PR is correctly updated :-) See:

Before:


After:

**Comment by danielzhe** [comment]
I found an issue.
If you have 2 structures and then drag one inside to another, looks like the visual indicator doesn't work anymore. Also some strange issues happened while playing with drag and drop, but I'm not sure if it's related to this PR. See the gif:
issue

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListHighlightHelper.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    private void cleanBackgroundLine() {
+        final NodeList<Element> elementNodeList = querySelectorAll("." + LEVEL_BACKGROUND_LINE);
+        for (int i = 0; i < elementNodeList.length; i++) {
+            final Element e = elementNodeList.getAt(i);
+            e.parentNode.removeChild(e);
+        }
+    }
+
+    private int countDataTypes(final DataType dataType) {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListHighlightHelper.java]: I think the name is misleading here, we are actually counting count of entries? Am I right?

**Comment by karreiro** [comment]
@danielzhe @jomarko Thank you for the review. Both issues are fixed now :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListHighlightHelper.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListHighlightHelper.java]: Fixed! :-)

**Comment by jomarko** [comment]
@karreiro I found an issue dragging nested structure. See the [comment]( https://issues.redhat.com/browse/DROOLS-4485?focusedCommentId=13958061&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13958061)

**Comment by karreiro** [comment]
@jomarko Thanks for reporting. Fixed!

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by karreiro** [comment]
@jomarko Could you please take a look on this?

**Pull request merged**